packages/react-reconciler/src/ReactFiberCommitEffects.js

```diff
@@ -246,7 +302,9 @@ export function commitHookEffectListUnmount(
           const inst = effect.inst;
           const destroy = inst.destroy;
           if (destroy !== undefined) {
-            inst.destroy = undefined;
+            if (effect.kind === SimpleEffectKind) {
+              inst.destroy = undefined;
+            }
```

Comment:
if we run this for all effect kinds it is a problem? or is this just for gating purposes?

Reply:
I'm sharing the EffectInstance between the identity and update and forwarding that each time it's pushed again onto the queue, so we can't null it out as it would break the chain and become undefined, if that makes sense